src/main/java/dk/bec/polonez/reservationsystem/service/DiscountService.java

```diff
@@ -1,52 +1,65 @@
 import dk.bec.polonez.reservationsystem.model.Discount;
-import dk.bec.polonez.reservationsystem.dto.DiscountDto;
+import dk.bec.polonez.reservationsystem.dto.offerDto.CreateDiscountDto;
+import dk.bec.polonez.reservationsystem.model.Offer;
 import dk.bec.polonez.reservationsystem.repository.DiscountRepository;
+import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
 import org.springframework.web.server.ResponseStatusException;
 
 import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Service
 public class DiscountService {
 
     private final DiscountRepository discountRepository;
-    @Autowired
+    private final ModelMapper modelMapper;
+
+
     public DiscountService(DiscountRepository discountRepository) {
         this.discountRepository = discountRepository;
+        this.modelMapper = new ModelMapper();
     }
 
-    public int getByCode(String code){
-        return discountRepository.getByCode(code);
+    public DiscountDto getByCode(String code) {
+        Discount discount = discountRepository.findDiscountByCode(code);
+        return modelMapper.map(discount, DiscountDto.class);
     }
 
-    public Discount findById(int id){
-        Optional<Discount> optionalDiscount = discountRepository.findById(id);
-        return optionalDiscount.
-                orElseThrow(()->new ResponseStatusException(HttpStatus.NOT_FOUND));
+    public DiscountDto getById(long id) {
+        Discount discount = discountRepository.getById(id);
+        return modelMapper.map(discount, DiscountDto.class);
     }
-
-    public List<Discount> getAll(){
-        return discountRepository.findAll();
+    public List<DiscountDto> getAll() {
+        List<Discount> discounts = discountRepository.findAll();
+        return discounts.stream()
+                .map(discount -> modelMapper.map(discount, DiscountDto.class))
+                .collect(Collectors.toList());
     }
 
-    public void deleteDiscount(int id){
-        discountRepository.deleteById(id);
+    public DiscountDto deleteDiscount(long id) {
+        Discount discountToDelete = discountRepository.getById(id);
+        discountRepository.delete(discountToDelete);
+        return modelMapper.map(discountToDelete, DiscountDto.class);
     }
 
-    public void updateDiscount(int id, DiscountDto updateDiscount){
-        Discount discountToUpdate = discountRepository.getById(id);
-
-        discountToUpdate.setCode(updateDiscount.getCode());
-        discountToUpdate.setDateFrom(updateDiscount.getDateFrom());
-        discountToUpdate.setDateTo(updateDiscount.getDateTo());
-        discountToUpdate.setName(updateDiscount.getName());
+    public DiscountDto updateDiscount(long id, CreateDiscountDto discountDto) {
+        Discount discount = discountRepository.getById(id);
+        modelMapper.map(discountDto, discount);
```

Comment:
**warn** Will discountId also be updated in line 54? Shouldn't id be unchange'able?

Reply:
No, it wont change, because CreateDiscountDto doesn't have id attribute.